packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/timing/TimingModuleTest.kt

```diff
@@ -0,0 +1,265 @@
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostPause()
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(mJSTimersMock)
+    reset(mJSTimersMock)
+    mTimingModule.onHostResume()
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+  }
+
+  @Test
+  fun testHeadlessJsTaskInBackground() {
+    mTimingModule.onHostPause()
+    mTimingModule.onHeadlessJsTaskStart(42)
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskInForeground() {
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskStart(42)
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostPause()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskIntertwine() {
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskStart(42)
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    mTimingModule.onHostPause()
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostPause()
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testSetTimeoutZero() {
+    mTimingModule.createTimer(100.0, 0.0, 0.0, false)
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(100.0))
+  }
+
+  @Test
+  fun testActiveTimersInRange() {
+    mTimingModule.onHostResume()
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse // Repeating
+    mTimingModule.createTimer(42.0, 150.0, 0.0, false)
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse // Out of range
+    assertThat(mTimingModule.hasActiveTimersInRange(200)).isTrue // In range
+  }
+
+  @Test
+  @Ignore("Failing due to failed invocation verification") // TODO T13905097
+  fun testIdleCallback() {
+    mTimingModule.onHostResume()
+    mTimingModule.setSendIdleEvents(true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callIdleCallbacks(SystemClock.currentTimeMillis().toDouble())
```

Comment:
This verification fails for some reason. I've decided that try to fix it is out of scope for this PR so will attempt to fix it in a subsequent PR.

Reply:
Did it fail previously, when it was in Java?

Reply:
@rshest yes the whole test suite was suppressed when in Java so this is net positive.